addons/payment_stripe/controllers/main.py

```diff
@@ -19,9 +18,45 @@
 
 class StripeController(http.Controller):
     _checkout_return_url = '/payment/stripe/checkout_return'
+    _onboarding_return_url = '/payment/stripe/onboarding/return/'
+    _onboarding_refresh_url = '/payment/stripe/onboarding/refresh/'
     _validation_return_url = '/payment/stripe/validation_return'
     WEBHOOK_AGE_TOLERANCE = 10*60  # seconds
 
+    @http.route(
+        _onboarding_return_url + '<int:acquirer_id>/<int:menu_id>', type='http', auth='public'
+    )
+    def stripe_onboarding_return(self, acquirer_id, menu_id):
+        """Stripe return URL
+
+            This route is used by Stripe to return after or during the onboarding.
+            Once called, the controller redirects to the payment acquirer.
+
+            :returns: Redirection to Stripe Acquirer Form
+        """
+        action = request.env.ref(
+            'payment_stripe.action_payment_acquirer_onboarding', raise_if_not_found=False
+        )
+        return request.redirect('/web?#action=%s&id=%s&menu_id=%s' % (
+            action and action.id or 'payment.action_payment_acquirer', acquirer_id, menu_id
+        ))
+
+    @http.route(
+        _onboarding_refresh_url + '<int:acquirer_id>/<account_id>/<int:menu_id>', type='http',
+        auth='public'
```

Comment:
either
```suggestion
        methods=['GET'], auth='public'
```
or
```suggestion
        auth='public', csrf=False
```
Maybe go directly for GET since it's the method that we expect, and it would solve or session cookie question once and for all.

Reply:
The first sounds better ofc